commit b5b3aadf40b5a94d79b4ec3f1d3a1892eba01f6a
Author: wm4 <wm4@nowhere>
Date:   Sat Apr 29 15:09:26 2017 +0200

vdpau, vaapi: unref libavutil device ref

It's a simple memory leak. (The API objects were destroyed anyway.)

diff --git a/video/vaapi.c b/video/vaapi.c
index 171ccdfcb..76a197cca 100644
--- a/video/vaapi.c
+++ b/video/vaapi.c
@@ -224,6 +224,8 @@ error:
 void va_destroy(struct mp_vaapi_ctx *ctx)
 {
     if (ctx) {
+        av_buffer_unref(&ctx->av_device_ref);
+
         if (ctx->display)
             vaTerminate(ctx->display);

diff --git a/video/vdpau.c b/video/vdpau.c
index 889505324..fa79e9bc6 100644
--- a/video/vdpau.c
+++ b/video/vdpau.c
@@ -481,6 +481,8 @@ void mp_vdpau_destroy(struct mp_vdpau_ctx *ctx)
         CHECK_VDP_WARNING(ctx, "Error when calling vdp_output_surface_destroy");
     }

+    av_buffer_unref(&ctx->av_device_ref);
+
     if (ctx->preemption_obj != VDP_INVALID_HANDLE) {
         vdp_st = vdp->output_surface_destroy(ctx->preemption_obj);
         CHECK_VDP_WARNING(ctx, "Error when calling vdp_output_surface_destroy");


commit bc3365b1693ce2a2c795a489c64738a42d5c7e6b
Author: Raúl Peñacoba <raul.mikaop.zelda@gmail.com>
Date:   Wed May 10 22:21:16 2017 +0200

video/out/wayland/buffer.c: Add missing free()

diff --git a/video/out/wayland/buffer.c b/video/out/wayland/buffer.c
index aae590eba..dce3ca433 100644
--- a/video/out/wayland/buffer.c
+++ b/video/out/wayland/buffer.c
@@ -40,13 +40,16 @@ shm_buffer_t* shm_buffer_create(uint32_t width,
     shm_buffer_t *buffer = calloc(1, sizeof(shm_buffer_t));
     int fd = memfile_create(size);

-    if (fd < 0)
+    if (fd < 0) {
+        free(buffer);
         return NULL;
+    }

     buffer->data = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0);

     if (buffer->data == MAP_FAILED) {
         close(fd);
+        free(buffer);
         return NULL;
     }

